blocks/bricks/attention.py

```diff
@@ -605,8 +605,11 @@ def take_glimpses(self, **kwargs):
             kwargs.pop(self.preprocessed_attended_name, None),
             kwargs.pop(self.attended_mask_name, None),
             **dict_union(states, glimpses_needed))
-        if kwargs:
-            raise ValueError("extra args to take_glimpses: {}".format(kwargs))
+        # The following check is too excessive.
+        # there could be context information which is needed by the transition
+        # apply, but is also passed to this method.
+        # if kwargs:
+        #    raise ValueError("extra args to take_glimpses: {}".format(kwargs))
```

Comment:
Can you provide an example? `take_glimpses` is called in `do_apply` where we provide only states, glimpses, and attended stuff. So it shouldn't be any variables for `transition.apply` here.